app/controllers/api/enterprise_attachment_controller.rb

```diff
@@ -0,0 +1,37 @@
+module Api
+  class EnterpriseAttachmentController < BaseController
+    class MissingImplementationError < StandardError; end
+    class UnknownEnterpriseAuthorizationActionError < StandardError; end
+
+    before_filter :load_enterprise
+
+    respond_to :json
+
+    def destroy
+      return respond_with_conflict(error: destroy_attachment_does_not_exist_error_message) unless @enterprise.send("#{attachment_name}?")
```

Comment:
we should favor `public_send` over `send` as the former respects private methods while the latter does not.

Reply:
Maybe this can be a convention to be added here? https://github.com/openfoodfoundation/openfoodnetwork/wiki/Code-Conventions

Reply:
nice one @luisramos0 ! There must be a Rubocop's cop for that though...

Reply:
😎 https://github.com/rubocop-hq/rubocop/pull/2081 Rubocop never let's you down

Reply:
Noting that, @sauloperez. I've never used `public_send` before, actually.